packages/ember-toucan-core/src/components/form/controls/character-count.ts

```diff
@@ -0,0 +1,19 @@
+import Component from '@glimmer/component';
+
+export interface ToucanFormControlCharacterCountComponentSignature {
+  Element: HTMLSpanElement;
+  Args: {
+    /**
+     * The current number of characters inside the input
+     */
+    current?: number;
```

Comment:
Are these args really optional? As you don't seem to supply a default value, so omitting them would probably render as `undefined / undefined`? I think we could keep `current` as optional, by defaulting it to zero. But `max` is probably required, as we cannot assume any default value reasonably, right?

If we allow some to be optional, then a test case for this behaviour would also be helpful!

Reply:
I tried making @max required, and then @current optional with something like

```javascript
// in the backing class
@tracked current = this.args.current ?? 0;

// in the constructor
this.current = this.args.current;
```

However, the @current arg was not getting updated in the test and so the test was breaking. So I went back to making `@current` required.

I would THINK that this would work, given that's how I think this works in React, but it seems like the constructor is not called in Ember everytime an argument changes??

Reply:
Make both required.